gantt_edit.php

```diff
@@ -85,18 +92,37 @@
 		}
 		if ( in_array( '', $infoCategory, true ) )
 		{
-			exit;
+			$validationMsg = 'At least one category is incomplete.';
+			break;
 		}
 		$listCategories[] = $infoCategory;
 	}
 	if ( count( $listLabels ) == 0 || count( $listCategories ) == 0 )
+	{
+		$validationMsg = 'At least one label and one category is required.';
+	}
+	if ( isset( $_SERVER['HTTP_X_RC_ADVREP_GANTTCHK'] ) )
+	{
+		header( 'Content-Type: application/json' );
+		if ( $validationMsg == '' )
+		{
+			echo 'true';
+		}
+		else
+		{
+			echo json_encode( $validationMsg );
+		}
+		exit;
+	}
+	if ( $validationMsg != '' )
 	{
 		exit;
```

Comment:
Should error message display here? Also are we not returning to report edit page so error can be fixed.